### PR TITLE
LPS-139865 Fix PagintationBarTag server side rendering

### PR DESCRIPTION
**Pull request by julien**
The PagintationBarTag wasn't rendering exactly the same html
as it's corresponding Clay component, this change fixes that.

**Test plan**:
- Fetch this branch
- Deploy the `frontend-taglib-clay` OSGi module
- Assert that the PaginationBarTag renders the same html
  as the Clay component

### CODE REVIEW

**Comment by julien** [comment]
@john-co I had a look at the CI errors and they don't seem related, but could you confirm when you get a chance? Thanks.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (nhpatt) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:273]: You can just cast one and everything will be casted to double but this is Ok and maybe the SF is enforcing it (hope not), let's leave it this way.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (nhpatt) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:295]: Typically we would define this variables just before using them to see more clearly the logic.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (nhpatt) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:303]: Could this be null?

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (julien) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:273]: I was getting different results than in the [React component](https://github.com/liferay/clay/blob/79801c41aa12f40685b2118467c79421ce9cd492/packages/clay-pagination-bar/src/PaginationBarWithBasicItems.tsx#L170) without these casts, but I can try again.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (julien) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:303]: I don't think it would

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (julien) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:295]: Sure, let me reorganize that.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (julien) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:303]: It shouldn't be null if that is what you mean, but if there's a cleaner way to do it, I'm always interested.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (nhpatt) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:273]: Don't, no need.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (nhpatt) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:303]: I was meaning that, it there is a case where totalItems is null. If it was you could safe guard with an if or use an Optional.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (julien) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java]: Should look better in https://github.com/liferay-frontend/liferay-portal/pull/1501/commits/ae06adc4a85b942d6add42f8202812a7d3fd4d5a

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (matuzalemsteles) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:360]: @julien which case does a pagination need to be configured to fall into this use case? Maybe we also have to add in the React component?

**Review by matuzalemsteles — APPROVED** [review]
LGTM! Good job, the logic is very similar to the JS implementation.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (matuzalemsteles) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:284]: `totalItems` should be a mandatory property to render the minimo but I'm not sure if we add an initial value even when it's not passed. I imagine that because it is mandatory in taglib it must fail at compile time.

**Comment by john-co** [comment]
Reviewed. Test failures are unrelated. :heavy_check_mark: 
- 2% flaky test batch based on recent test history: [lpkg-base-jdk8/0](https://testray.liferay.com/home/-/testray/cases/1328274400)

cc/ @vicnate5, is this something your team could look more into? I'm not sure who owns the test lpkg batch.

**Comment by john-co** [comment]
ci:test:relevant

**Review by nhpatt — APPROVED** [review]

**Comment by nhpatt** [comment]
LGTM :)

One possible improvement would be trying to remove the new level of indentation, I saw that using an early return wasn't possible but maybe extracting to methods would have been removed (didn't check).

**Comment by john-co** [comment]
ci:reevaluate:1317401_645

**Comment by john-co** [comment]
Reviewed. Test failures are unrelated. :heavy_check_mark: 
commerce tests recently started failing. ci reevaluate should be able to mark them as common for this PR.

**Comment by john-co** [comment]
@iovaneale @Pietro69 @lorenzolapescara the unique failing test is unrelated and has recently been failing upstream.

It's not being marked as common failure correctly because the test hasn't been run on acceptance recently. I suggest setting both environment and portal acceptance property to either "true" or "quarantine".

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (julien) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:360]: @matuzalemsteles I'll have to check because I'm not 100% but I think it was with these values.

```
activePage = 15
ellipsisBuffer = 2
totalItems = 17
```

About adding that to the react component, I'm not sure it's needed either, Java was throwing an `IndexOutOfBoundsException` but JavaScript (well TypeScript) doesn't even complain

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (julien) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:284]: `totalItems` is already a [required](https://github.com/liferay/liferay-portal/blob/7da6852989d3c87e1d763eca20bfde198c1a8afe/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/liferay-clay.tld#L2221-L2226) attribute. In the React component, it looks like the "prop" [isn't optional](https://github.com/liferay/clay/blob/386008b990fc2670b3944b538de29531d49c6d59/packages/clay-pagination/src/PaginationWithBasicItems.tsx#L60) either.

**Comment by pietrobocale** [comment]
@john-co We realize that we have a blocker (https://issues.liferay.com/browse/COMMERCE-7349). This issue is already fixed, we need only wait for the merge.

**Comment by julien** [comment]
> One possible improvement would be trying to remove the new level of indentation, I saw that using an early return wasn't possible but maybe extracting to methods would have been removed (didn't check).

@nhpatt sorry I don't understand this comment but I know that when we worked on the [ManagementToolbarTag](https://github.com/liferay/liferay-portal/blob/7da6852989d3c87e1d763eca20bfde198c1a8afe/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ManagementToolbarTag.java) we also thought there was a lot of markup and "nesting" (I had originally had placed some of that in code in separate methods), but we decided to keep all the rendering logic in the `processStartTag` method.

**Comment by julien** [comment]
ci:test:relevant

**Comment by nhpatt** [comment]
@julien Got it :) LGTM :D

**Review by nhpatt — APPROVED** [review]

**Comment by magjed4289** [comment]
ci:test:bundle

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java
Comment (matuzalemsteles) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/PaginationBarTag.java:360]: Hmm I see, no problem.

**Comment by magjed4289** [comment]
**LocalFile.CommerceSmoke#PublishSimpleProduct** fail unrelated to the PR
**LocalFile.CPCommerceProducts#ProductPublishing** fail unrelated to the PR
**LocalFile.CommerceAccelerators#InitializeNewMiniumSite** fail unrelated to the PR
**LocalFile.CommerceAccelerators#InitializeNewSpeedwellSite** fail unrelated to the PR

All tests fail at least 3 out of 200 latest executions.

**Comment by john-co** [comment]
I'll manually forward this since looks like got double approved.

**Comment by julien** [comment]
Thanks a lot @john-co

**Comment by john-co** [comment]
Pushed to https://github.com/brianchandotcom/liferay-portal/pull/107900

**Comment by john-co** [comment]
> @john-co We realize that we have a blocker (https://issues.liferay.com/browse/COMMERCE-7349). This issue is already fixed, we need only wait for the merge.

Thanks @pietrobocale, next time can you also send a PR to BChan directly to quarantine the test while the fix was being worked on. Some of our modules track environment acceptance for `ci:test:relevant` and it blocks our `ci:forward` attempts. See https://github.com/liferay-frontend/liferay-portal/pull/1505#issuecomment-929491556

**Comment by john-co** [comment]
@pietrobocale , taking a step back I'm going to patch it up on FI side in case something like this happens again: https://github.com/liferay-frontend/liferay-portal/pull/1509

**Pull request closed without merging**
